src/Stack.php

```diff
@@ -46,6 +47,22 @@ public function prepend(ServerMiddlewareInterface $middleware)
     }
 
     /**
+     * Process an incoming server request and return a response, optionally delegating
+     * to the next middleware component to create the response.
+     *
+     * @param ServerRequestInterface $request
+     * @param DelegateInterface $nextContanierDelegate
+     *
+     * @return ResponseInterface
+     */
+    public function process(ServerRequestInterface $request, DelegateInterface $nextContanierDelegate)
+    {
+        $delegate = new Delegate($this->middleware, new DelegateToCallableAdapter($nextContanierDelegate));
```

Comment:
Okay, we're getting off track here. I want to see two changes, made as separate PRs:

- move `$default` to the constructor as the first argument
- implement `ServerMiddlewareInterface` in Stack